test/validator/size_test.dart

```diff
@@ -18,14 +18,56 @@ ValidatorCreator size(int size) {
 }
 
 main() {
-  setUp(d.validPackage.create);
+  test('considers a package valid if it is <= 100 MB', () async {
+    await d.validPackage.create();
 
-  test('considers a package valid if it is <= 100 MB', () {
     expectNoValidationError(size(100));
     expectNoValidationError(size(100 * math.pow(2, 20)));
   });
 
-  test('considers a package invalid if it is more than 100 MB', () {
-    expectValidationError(size(100 * math.pow(2, 20) + 1));
+  group('considers a package invalid if it is more than 100 MB', () {
+    test('package is not under source control and no .gitignore exists',
+        () async {
+      await d.validPackage.create();
+
+      var errorsMatcher = contains(equals(
+          "Your package is 100.0 MB. Hosted packages must be smaller than 100 MB."));
+      expectValidationError(size(100 * math.pow(2, 20) + 1),
+          errorsMatcher: errorsMatcher);
+    });
+
+    test('package is not under source control and .gitignore exists', () async {
+      await d.validPackage.create();
+      await d.dir(appPath, [d.file('.gitignore', 'ignored')]).create();
+
+      var errorsMatcher = contains(allOf(
+          contains("Hosted packages must be smaller than 100 MB."),
+          contains(
+              "Your .gitignore has no effect since your project does not appear to be in version control.")));
+      expectValidationError(size(100 * math.pow(2, 20) + 1),
+          errorsMatcher: errorsMatcher);
+    });
+
+    test('package is under source control and no .gitignore exists', () async {
+      await d.validPackage.create();
+      await d.git(appPath).create();
```

Comment:
There are tests that call `ensureGit` and tests without it. Which way is more preferred?